workers/social/lib/social/models/group/groupdata.coffee

```diff
@@ -15,5 +19,83 @@ module.exports = class JGroupData extends Module
         type      : String
         validate  : require('../name').validateName
         set       : (value) -> value.toLowerCase()
+      payload     : Object
+
+    sharedMethods :
+      static      :
+        fetchByKey: [
+          (signature String, Function)
+        ]
+
+  @create = (slug, callback) ->
+    data = new JGroupData { slug }
+    data.save (err) ->
+      # this happens if socialapi creates the document in between fetch/create
+      # operations.
+      if err?.code is 11000 # duplicate key error
+        return JGroupData.fetchData slug, callback
+
+      return callback err  if err
+      callback null, data
+
+  @fetchData: (slug, callback) ->
+    return callback new Error 'slug is required' unless slug
+
+    JGroupData.one { slug }, (err, data) ->
+      return callback err  if err
+      return callback null, data if data
+
+      JGroupData.create slug, callback
+
+  # fetchByKey fetches given path from GroupData if only they are available
+  # within the group.
+  #
+  # @param {String} path
+  #   Provide the path you would like to fetch.
+  #
+  # @return {Object} persisted value.
+  @fetchByKey = secure (client, path, callback) ->
```

Comment:
Yes that is right, but please see line 64. It will fetch the current group's data, not from the ones you don't belong to.